api/src/pcapi/routes/backoffice/admin/bo_users_blueprint.py

```diff
@@ -34,6 +35,13 @@
 )
 
 
+def _get_bo_user_query(user_id: int) -> BaseQuery:
+    return users_models.User.query.outerjoin(users_models.User.backoffice_profile).filter(
+        users_models.User.id == user_id,
+        perm_models.BackOfficeUserProfile.id.is_not(None),
+    )
```

Comment:
[question] Pourquoi faire un **outer** join et vérifier l'existence de `BackOfficeUserProfile.id`, plutôt de que juste faire un **inner** join (qui a la même garantie) ?

```suggestion
    return users_models.User.query.join(users_models.User.backoffice_profile).filter_by(id=user_id)
```

Reply:
Bien vu, j'ai en fait pris en référence la fonction de recherche qui fait déjà cette jointure ainsi, je modifie donc les deux, merci :)

Reply:
@dbaty j'ai poussé la correction, mais aussi la même chose sur les endpoints pro ; je n'ai pas touché aux endpoints jeunes et grand public car cela mérite discussion : tout compte pro est par exemple aussi un compte grand public, et peut être éventuellement sur un parcours bénéficiaire en même temps : on peut avoir besoin d'agir côté jeune dessus.